lib/stockboy/configurator.rb

```diff
@@ -29,10 +29,16 @@ class Configurator
     # @overload new(&block)
     #   Evaluate DSL in a block
     #
-    def initialize(dsl='', file=__FILE__, &block)
+    def initialize(dsl='', file=__FILE__, inject_variables={}, &block)
       @config = {}
       @config[:triggers] = Hash.new { |hash, key| hash[key] = [] }
       @config[:filters] = {}
+
+      inject_variables.each do |k, v|
+        raise DSLVariableCollision if instance_variable_defined?("@#{k}")
+        instance_variable_set("@#{k}", v)
+      end
+
```

Comment:
Instead of creating individual variables here, let's define the entire group in a hash, I think this would make sense to access in the DSL as:

```
provider :file do
  file_name env[:file_name]
end
```

so there is no worry of collision, and the available keys are available there to be inspected.